src/main/java/com/watchtogether/watchtogether/member/service/MemberService.java

```diff
@@ -0,0 +1,49 @@
+package com.watchtogether.watchtogether.member.service;
+
+import com.watchtogether.watchtogether.exception.custom.MemberIdAlreadyUseException;
+import com.watchtogether.watchtogether.member.dto.MemberJoinDto;
+import com.watchtogether.watchtogether.member.entity.Member;
+import com.watchtogether.watchtogether.member.entity.Role;
+import com.watchtogether.watchtogether.member.repository.MemberRepository;
+import java.time.LocalDate;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class MemberService {
+
+  private final MemberRepository memberRepository;
+  private final PasswordEncoder passwordEncoder;
+
+  public Member joinMember(MemberJoinDto dto) {
+
+    // 이미 사용중인 ID 라면 예외 발생
+    if (memberRepository.findByMemberId(dto.getMemberId()).isPresent()) {
+      throw new MemberIdAlreadyUseException("이미 사용중인 ID 입니다.");
+    }
+
+    // 저장을 위한 Member 객체 초기화
+    Member member = Member.builder()
+        .memberId(dto.getMemberId())
+        // 비밀번호 암호화
+        .password(passwordEncoder.encode(dto.getPassword()))
+        // 초기 포인트는 0으로 초기화
+        .point(0)
+        .name(dto.getName())
+        .email(dto.getEmail())
+        .tel(dto.getTel())
+        .role(Role.ROLE_USER)
```

Comment:
아래의 파트너 권한 부분을 이쪽으로 옮겨서 아래처럼 좀더 간편화 시켜도 좋을 것 같아요. 
일반적으로 개발시에 한번 할당한 값을 다시 재할당하는 개발은 지양되고 있긴합니다. (final화하여 선언)
따라서 요런식으로 개발하셔도 좋을 것 같아요
```suggestion
        .role(dto.isPartner()? Role.ROLE_PARTNER : Role.ROLE_USER)
```

Reply:
저도 말씀해주신 방식을 생각하고 있었는데 IF 문을 고민하다 코드가 가독성이 떨어질것 같아 마지막에 수정하는 방식으로 작업한거였습니다.
삼항연산자를 깜빡하고 있었습니다. 좋은 의견 감사드립니다.